py/tests/integration/test_spend_tx.py

```diff
@@ -143,6 +150,64 @@ def test_not_enough_tokens():
     common.stop_node(bob_node)
     shutil.rmtree(root_dir)
 
+def test_send_by_name():
+    # Bob registers a name 'bob.aet'
+    # Alice should be able to send tokens to Bob using that name
+    test_settings = settings["test_send_by_name"]
+    coinbase_reward = common.coinbase_reward() 
+    (root_dir, node, ext_api, top) = setup_node_with_tokens(test_settings, "miner") 
+    int_api = common.internal_api(node)
+
+    alice_private_key = keys.new_private()
+    alice_public_key = keys.public_key(alice_private_key)
+    alice_address = keys.address(alice_public_key)
+
+    bob_private_key = keys.new_private()
+    bob_public_key = keys.public_key(bob_private_key)
+    bob_address = keys.address(bob_public_key)
+
+    # initial balances - amounts that the miner should send them
+    alice_init_balance = test_settings["send_tokens"]["alice"]
+    bob_init_balance = test_settings["send_tokens"]["bob"]
+
+    # populate accounts with tokens
+    miner_send_tokens(alice_address, alice_init_balance, int_api, ext_api)
+    miner_send_tokens(bob_address, bob_init_balance, int_api, ext_api)
+
+    # validate balances
+    alice_balance0 = common.get_account_balance(int_api, pub_key=alice_address).balance
+    bob_balance0 = common.get_account_balance(int_api, pub_key=bob_address).balance
+
+    assert_equals(alice_balance0, alice_init_balance)
+    assert_equals(bob_balance0, bob_init_balance)
+    print("Alice balance is " + str(alice_balance0))
+    print("Bob balance is " + str(bob_balance0))
+    print("Bob address is " + bob_address)
+
+    bob_name = test_settings["name_register"]["name"]
+    register_name(bob_name, bob_address, ext_api, bob_private_key)
+
+    print("Bob has registered " + bob_name)
+    bob_balance1 = common.get_account_balance(int_api, pub_key=bob_address).balance
+    print("Bob balance is " + str(bob_balance1))
+
+    tokens_to_send = test_settings["spend_tx"]["amount"]
+    print("Alice is about to send " + str(tokens_to_send) + " to " + bob_name)
+    send_tokens_to_name(bob_name, tokens_to_send, alice_address, alice_private_key, ext_api)
+
+    # validate balances
+    alice_balance2 = common.get_account_balance(int_api, pub_key=alice_address).balance
+    bob_balance2 = common.get_account_balance(int_api, pub_key=bob_address).balance
+
+    print("Alice balance is " + str(alice_balance2))
+    print("Bob balance is " + str(bob_balance2))
+
+    assert_equals(alice_balance2, alice_balance0 - tokens_to_send - 1)
```

Comment:
I understand this 1 is meant to be test_settings["spend_tx"]["fee"] (had it been defined).

Reply:
Yes, I though I'd get a cleaner interface if remove the fee from the settting